File: Python/5-Formacao_vetor_unico.py
# O algoritmo encontra a intersecção de dois vetores A e B. Se não houver intersecção, exibe "Intersecção vazia".

def intersecao_vetores(A, B):
    IT = []
    C = 0
    for i in range(len(A)):
        for j in range(len(B)):
            if A[i] == B[j]:
                C += 1
                IT.append(A[i])
                break
    if C == 0:
        print("Intersecção vazia")
    else:
        print("Intersecção:", IT)

# # Exemplo de uso:
# A = [1, 2, 3]
# B = [2, 3, 4]
# intersecao_vetores(A, B)

# Saída: Intersecção: [2, 3]

File: Java_Fragoso/src/5-Formacao_Vetor_unico.java
//O algoritmo encontra a intersecção de dois vetores A e B. Se não houver intersecção, exibe "Intersecção vazia".
import java.util.ArrayList;
import java.util.Arrays;

public class Formacao_Vetor_unico {
    public static void main(String[] args){
        int[] a = {1,2,3};
        int[] b = {2,3,4};

        ArrayList<Integer> IT = new ArrayList<Integer>();
        int C = 0;
        for(int i = 0; i < a.length; i++){
            for(int j = 0; j < b.length; j++){
                if(a[i] == b[j]){
                    C += 1;
                    IT.add(a[i]);
                }
            }
        }
        if(C == 0){
            System.out.println("Intersecção vazia");
        }
        else{
            System.out.println("Intersecção: " + IT);
        }
    }
}